src/controllers/searchController.js

```diff
@@ -0,0 +1,132 @@
+import Models from '../db/models';
+import errorResponse from '../helpers/index';
+
+const { User, Article, Tag } = Models;
+
+/**
+ * @description class representing Article Controller
+ * @class SearchController
+ */
+class SearchController {
+  /**
+   * @description - This method is responsible for creating new articles
+   * @static
+   * @param {object} request - Request sent to the router
+   * @param {object} response - Response sent from the controller
+   * @returns {object} - object representing response messages
+   * @memberof SearchController
+   */
+  static async searchByAuthor(request, response) {
+    const authorName = request.params.author;
+    try {
+      const author = await User.findOne({
+        where: {
+          username: authorName
+        },
+        include: [
+          {
+            model: Article,
+            as: 'articles',
+            attributes: ['slug', 'title', 'description', 'body', 'imgUrl']
+          }
+        ],
+        attributes: ['id', 'username', 'image', 'email', 'image']
+      });
+      if (author) {
+        if (author.dataValues.articles.length < 1) {
```

Comment:
what happens when `author.dataValues.articles.length === 0` versus `author.dataValues.articles.length < 1`?

Reply:
@marcdomain thanks for your observation. `author.dataValues.articles.length === 0` and `author.dataValues.articles.length < 1` returns the same result which is `true`. They are practically the same thing.